src/org/zaproxy/zap/extension/pscanrulesAlpha/UsernameIdorScanner.java

```diff
@@ -0,0 +1,205 @@
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.users.ExtensionUserManagement;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.users.User;
+
+public class UsernameIdorScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.usernameidor.";
+	private static final int PLUGIN_ID = 10057;
+
+	private PassiveScanThread parent = null;
+	private static final Logger LOGGER = Logger.getLogger(UsernameIdorScanner.class);
+
+	private List<User> usersList = new ArrayList<User>();
+
+	private ExtensionUserManagement extUserMgmt;
+
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	private boolean shouldContinue() {
+		return !getUsers().isEmpty(); // Should continue if not empty
+	}
+
+	private List<User> getUsers() {
+//		setUsers("guest");// Uncomment for testing/debugging purposes [guest MD5: 084e0343a0486ff05530df6c705c8bb4]
+		if (usersList.isEmpty()) {
+			for (Context context : Model.getSingleton().getSession().getContexts()) {
+				usersList.addAll(extUserMgmt.getContextUserAuthManager(context.getIndex()).getUsers());
+			}
+		}
+		return usersList;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		extUserMgmt = getExtensionUserManagement();
+
+		if (!shouldContinue()) {
+			if (LOGGER.isDebugEnabled()) {
+				LOGGER.debug("There does not appear to be any contexts with configured users.");
+			}
+			return;
+		}
+
+		long start = System.currentTimeMillis();
+
+		String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+		String username;
+
+		for (User user : getUsers()) {
+			username = user.getName();
+			Map<String, String> hashes = new HashMap<String, String>();
+			hashes.put("MD2", DigestUtils.md2Hex(username));
+			hashes.put("MD5", DigestUtils.md5Hex(username));
+			hashes.put("SHA1", DigestUtils.sha1Hex(username));
+			hashes.put("SHA256", DigestUtils.sha256Hex(username));
+			hashes.put("SHA384", DigestUtils.sha384Hex(username));
+			hashes.put("SHA512", DigestUtils.sha512Hex(username));
+			Iterator<Entry<String, String>> iter = hashes.entrySet().iterator();
+			while (iter.hasNext()) {
+				Map.Entry<String, String> pair = iter.next();
+				String hash = pair.getValue();
+				ContentsMatcher hashPatternMatcher = new PatternContentsMatcher(Pattern.compile(hash, Pattern.CASE_INSENSITIVE));
+				String evidence = hashPatternMatcher.match(response);
+				if (evidence != null) {
+					this.raiseAlert(username, evidence, pair.getKey(), id, msg);
+				}
+				iter.remove();

```

Comment:
Could be removed? The iteration will keep going anyway, and it could use enhanced for instead.